### PR TITLE
BL-764, handle book on network drive

### PR DESCRIPTION
**Pull request by JohnThomson**

### CODE REVIEW

File: src/BloomExe/web/ServerBase.cs [hunk 1]
```diff
@@ -17,6 +18,8 @@ namespace Bloom.web
 {
 	public abstract class ServerBase : IDisposable
 	{
+		internal const string SlashBloomSlash = "/bloom/";
```
Comment (hatton) [inline src/BloomExe/web/ServerBase.cs]: Two thoughts. One, why is this /bloom/ needed? Could you put a comment there? Two, this is an unusual variable name that just echos, etically, its value. Could it be something like "bloomPathPrefix"?

**Comment by hatton** [comment]
Can you think of a way of refactoring so that unit tests could be added for various kinds of paths?

File: src/BloomExe/web/ServerBase.cs (hunk 1 above)
Comment (JohnThomson) [inline src/BloomExe/web/ServerBase.cs]: Problem is, I don't really know why it was done. All our localhost: urls include it so we need to strip it off...possibly this constant should be more widely used, I just needed it in two places so I defined a constant. I'll at least improve the name.

File: src/BloomExe/Publish/MakePdfUsingGeckofxHtmlToPdfProgram.cs [hunk 2]
```diff
@@ -38,7 +38,8 @@ class MakePdfUsingGeckofxHtmlToPdfProgram
 			var runner = new CommandLineRunner();
 			string exePath;
 			var bldr = new StringBuilder();
-			var loc = Assembly.GetExecutingAssembly().Location;
+			// Codebase is reliable even when Resharper copies the EXE somewhere else for testing.
+			var loc = Assembly.GetExecutingAssembly().CodeBase.Substring("file:///".Length);
```
Comment (JohnThomson) [inline src/BloomExe/Publish/MakePdfUsingGeckofxHtmlToPdfProgram.cs]: This change is really nothing to do with this commit except I needed it to get unit tests passing locally with Resharper

File: src/BloomExe/web/ServerBase.cs [hunk 3]
```diff
@@ -8,6 +8,7 @@
 using System.Net;
 using System.Threading;
 using System.Windows.Forms;
+using Amazon.CloudSearch.Model.Internal.MarshallTransformations;
```
Comment (hatton) [inline src/BloomExe/web/ServerBase.cs]: John, is this using intentional?

Comment (JohnThomson) [inline src/BloomExe/web/ServerBase.cs]: I didn't add it...presumably Resharper did. No idea why. I'll see if it works without it.

**Pull request merged**